subxt/src/blocks/extrinsic_types.rs

```diff
@@ -326,8 +347,100 @@ where
+    }
+
+    /// Return only the bytes of the extra fields for this signed extrinsic.
+    ///
+    /// # Note
+    ///
+    /// Returns `None` if the extrinsic is not signed.
+    pub fn extra_bytes(&self) -> Option<&[u8]> {
+        self.signed
+            .as_ref()
+            .map(|e| &self.bytes[e.signature_end_idx..e.extra_end_idx])
+    }
+
+    /// Returns a reference to the signed extensions.
+    ///
+    /// Returns `None` if the extrinsic is not signed.
+    /// Returns `Some(Err(..))` if the extrinsic is singed but something went wrong decoding the signed extensions.
+    pub fn signed_extensions(&self) -> Option<Result<ExtrinsicSignedExtensions<T>, Error>> {
+        fn signed_extensions_or_err<'a, T: Config>(
+            extra_bytes: &'a [u8],
+            extra_ty_id: u32,
+            metadata: &'a Metadata,
+        ) -> Result<ExtrinsicSignedExtensions<'a, T>, Error> {
+            let extra_type = metadata
+                .types()
+                .resolve(extra_ty_id)
+                .ok_or(MetadataError::TypeNotFound(extra_ty_id))?;
+            // the type behind this is expected to be a tuple that holds all the individual signed extensions
+            let TypeDef::Tuple(extra_tuple_type_def) = &extra_type.type_def else {
+                return Err(Error::Other(
+                    "singed extra type def should be a tuple".into(),
+                ));
+            };
+
+            let mut signed_extensions: Vec<ExtrinsicSignedExtension<T>> = vec![];
+
+            let cursor: &mut &[u8] = &mut &extra_bytes[..];
+            let mut start_idx: usize = 0;
+            for field in extra_tuple_type_def.fields.iter() {
+                let ty_id = field.id;
+                let ty = metadata
+                    .types()
+                    .resolve(ty_id)
+                    .ok_or(MetadataError::TypeNotFound(ty_id))?;
+                let name = ty.path.segments.last().ok_or_else(|| Error::Other("signed extension path segments should contain the signed extension name as the last element".into()))?;
+                scale_decode::visitor::decode_with_visitor(
+                    cursor,
+                    ty_id,
+                    metadata.types(),
+                    scale_decode::visitor::IgnoreVisitor,
+                )
+                .map_err(|e| Error::Decode(e.into()))?;
+                let end_idx = extra_bytes.len() - cursor.len();
+                let bytes = &extra_bytes[start_idx..end_idx];
+                start_idx = end_idx;
+                signed_extensions.push(ExtrinsicSignedExtension {
+                    bytes,
+                    ty_id,
+                    name,
+                    metadata,
+                    phantom: PhantomData,
+                });
+            }
+
+            Ok(ExtrinsicSignedExtensions {
+                bytes: extra_bytes,
+                signed_extensions,
+                ty_id: extra_ty_id,
```

Comment:
We don't need this type ID; instead of accessing and checking this type above, prefer `metadata.extrinsic().signed_extensions()`. This gives a vec of signed extension details that you can iterate to decode each of them :)

Then that type ID also doesn't need to propagate to any other types etc; it's not needed here.